src/api/tokenList/TokenListApi.ts

```diff
@@ -13,21 +26,67 @@ export interface TokenList {
 export class TokenListApiImpl implements TokenList {
   public networkIds: number[]
   private _tokensByNetwork: { [networkId: number]: TokenDetails[] }
+  private _tokenAddressNetworkSet: Set<string>
 
   public constructor(networkIds: number[]) {
     this.networkIds = networkIds
 
     // Init the tokens by network
     this._tokensByNetwork = {}
+    this._tokenAddressNetworkSet = new Set<string>()
+
     networkIds.forEach(networkId => {
       // initial value
-      this._tokensByNetwork[networkId] = getTokensByNetwork(networkId)
+      const tokenList = getTokensByNetwork(networkId).concat(this.loadUserTokenList(networkId))
+      this._tokensByNetwork[networkId] = tokenList
+
+      tokenList.forEach(({ address }) => {
+        this._tokenAddressNetworkSet.add(
+          TokenListApiImpl.constructAddressNetworkKey({ tokenAddress: address, networkId }),
+        )
+      })
     })
   }
 
+  public hasToken(params: HasTokenParams): boolean {
+    return this._tokenAddressNetworkSet.has(TokenListApiImpl.constructAddressNetworkKey(params))
+  }
+
   public getTokens(networkId: number): TokenDetails[] {
     return this._tokensByNetwork[networkId] || []
   }
+
+  private static constructAddressNetworkKey({ tokenAddress, networkId }: HasTokenParams): string {
+    return tokenAddress.toLowerCase() + '|' + networkId
+  }
+
+  private static getUserTokenListName(networkId: number): string {
+    return 'USER_TOKEN_LIST_' + networkId
```

Comment:
Does it have to be static?

Reply:
what does it return if nothing is instantiated? would probably be `undefined` or would it error?

Reply:
why not static? I don't use static enough, let me use static

Reply:
what instantiated? `networkId`? TS takes care that we only pass in `number`